examples/exNDH1ser.cpp

```diff
@@ -0,0 +1,281 @@
+   BilinearForm *a = new BilinearForm(fespace);
+   MixedBilinearForm *a_NDH1 = new MixedBilinearForm(H1fespace, fespace);
+   if (pa)
+   {
+      a->SetAssemblyLevel(AssemblyLevel::PARTIAL);
+      a_NDH1->SetAssemblyLevel(AssemblyLevel::PARTIAL);
+   }
+
+   a->AddDomainIntegrator(new VectorFEMassIntegrator(*sigma));
+
+   a_NDH1->AddDomainIntegrator(new MixedVectorGradientIntegrator(*muinv));
+
+   // 7. Assemble the parallel bilinear form and the corresponding linear
+   //    system, applying any necessary transformations such as: parallel
+   //    assembly, eliminating boundary conditions, applying conforming
+   //    constraints for non-conforming AMR, static condensation, etc.
+   if (static_cond) { a->EnableStaticCondensation(); }
+
+   a->Assemble();
+   if (!pa) { a->Finalize(); }
+
+   a_NDH1->Assemble();
+   if (!pa) { a_NDH1->Finalize(); }
+
+   Vector B(fespace->GetTrueVSize());
+   Vector X(fespace->GetTrueVSize());
+
+   if (pa)
+   {
+      a_NDH1->Mult(p, x);
+      x.GetTrueDofs(B);
+   }
+   else
+   {
+      SparseMatrix& NDH1 = a_NDH1->SpMat();
+      Vector P(H1fespace->GetTrueVSize());
+      p.GetTrueDofs(P);
+
+      NDH1.Mult(P,B);
+   }
+
+   // 8. Define and apply a PCG solver for AX=B with no
+   //    preconditioner, in the full assembly case.
+   //    With partial assembly, use Jacobi preconditioner.
+
+   if (pa)
+   {
+      GridFunction diag_pa(fespace);
+      Vector tdiag_pa(fespace->GetTrueVSize());
+      a->AssembleDiagonal(tdiag_pa);
+      diag_pa.SetFromTrueDofs(tdiag_pa);
+
+      Array<int> ess_tdof_list;
+      OperatorJacobiSmoother Jacobi(diag_pa, ess_tdof_list, 1.0);
+
+      CGSolver cg;
+      cg.SetRelTol(1e-12);
+      cg.SetMaxIter(1000);
+      cg.SetPrintLevel(1);
+      cg.SetOperator(*a);
+      cg.SetPreconditioner(Jacobi);
+
+      GridFunction rhs(fespace);
+      rhs = x;
+      x = 0.0;
+      cg.Mult(rhs, x);
+   }
+   else
+   {
+      SparseMatrix& Amat = a->SpMat();
+      CGSolver cg;
+      cg.SetRelTol(1e-12);
+      cg.SetMaxIter(1000);
+      cg.SetPrintLevel(1);
+      cg.SetOperator(Amat);
+      cg.Mult(B, X);
+
+      x.SetFromTrueDofs(X);
+   }
+
+   // 9. Compute the same solution by applying the GradientInterpolator in H(curl).
+   
+   DiscreteLinearOperator grad(H1fespace, fespace);
+   grad.AddDomainInterpolator(new GradientInterpolator());
+   grad.Assemble();
+
+   GridFunction gradp(fespace);
+   grad.Mult(p, gradp);
+
+   // 10. Compute the projection of the exact grad p.
+   
+   GridFunction exact_gradp(fespace);
+   exact_gradp.ProjectCoefficient(gradp_coef);
+   
+   // 11. Compute and print the L^2 norm of the error.
+   {
+      double errSol = x.ComputeL2Error(gradp_coef);
+      double errInterp = gradp.ComputeL2Error(gradp_coef);
+      double errProj = exact_gradp.ComputeL2Error(gradp_coef);
+
+      cout << "\n Solution of (E_h,v) = (grad p,v) for E_h and v in H(curl): || E_h - grad p ||_{L^2} = " << errSol << '\n' << endl;
```

Comment:
These lines should be limited to 80 characters

Reply:
Done.